runtests.py

```diff
@@ -878,7 +878,7 @@ def setUp(self):
         from Cython.Compiler import Options
         self._saved_options = [
             (name, getattr(Options, name))
-            for name in ('warning_errors', 'clear_to_none', 'error_on_unknown_names', 'error_on_uninitialized')
+            for name in dir(Options) if not name.startswith("_")
```

Comment:
I actually kept it explicit at the time which options could be set in tests, but wouldn't object to making it dynamic if it gets in the way. Seems better to use `vars()`, though, instead of `dir()` + `getattr()`.

Reply:
Adding `cache_builtins` to runtests.py turned out to be a disaster because the contents of the builtins scope isn't reset between tests. I've rolled this back and added a very simple srctree test instead.